EIPS/eip-1319.md

```diff
@@ -166,6 +166,62 @@ release versions while allowing them to use any versioning schema they choose.
 
 Registries may offer more complex `read` APIs that manage requests for packages within a semver range or at `latest` etc. This EIP is agnostic about how tooling or registries might implement these. It recommends that registries implement [EIP 165](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-165.md) and avail themselves of resources to publish more complex interfaces such as [EIP 926](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-926.md).
 
+## EthPM URIs
+
+EthPM URIs are used to identify on-chain erc1319-compliant (ethPM) registries and their packages. The goal of this URI specification is to provide a human and machine readable string that can uniquely identify any package release across any registry. This simple datatype allows for quick, one-click installs and convenient transportion of ethpm packages between tooling/frameworks.
+
+### Format
+
+`scheme://registry_address:chain_id/package_name@package_version/json_pointer`
+
+#### `scheme`
+- Required
+- Must be one of `ethpm` or `erc1319`
+
+#### `registry_address`
+- Required
+- This **should** be an ENS name whenever possible, or a `0x`-prefixed, checksummed address. 
+
+#### `chain_id`
+- Optional
+- Integer representing the chain id on which the registry is located
+- If omitted, defaults to `1` (mainnet).
+
+#### `package_name`
+- Optional
+- String of the target package name
+
+#### `package_version`
+- Optional
+- String of the target package version
+- If the package version contains any [url unsafe characters](https://www.werockyourweb.com/url-escape-characters/), they **must** be safely escaped
+- Since semver is not strictly enforced by the ethpm spec, if the `package_version` is omitted from a uri, it is up to the tooling / framework to look up all available versions and ask the user which version they want to utilize.
```

Comment:
This would probably be better to use the `**should**` language.  Something like:

> if the `package_version` is omitted from a uri, tooling **should** avoid guessing in the face of any ambiguity and present the user with a choice from the available versions.